core/src/main/java/org/mule/runtime/core/exception/TemplateOnErrorHandler.java

```diff
@@ -110,11 +110,11 @@ protected MuleEvent processCatch(MuleEvent event, MessagingException exception)
         TransactionCoordination.getInstance().rollbackCurrentTransaction();
       } catch (Exception ex) {
         //Do nothing
+        logger.warn(ex.getMessage());
       }
 
-      event.setError(builder(exception).build());
       event.setMessage(MuleMessage.builder(event.getMessage()).exceptionPayload(new DefaultExceptionPayload(exception)).build());
-      return event;
+      throw exception;
```

Comment:
Why were this changes necessary? Seems like they might mess with the catch/rollback approaches.